### PR TITLE
[CARBONDATA-4001] Fix SI global sort load on partition table

### PR DESCRIPTION
**Pull request by ajantha-bhat**
### Why is this PR needed?
On a partition table,  when SI is created with global sort and when data is loaded. It shows 0 rows as main table query results  is 0 rows.
 
 ### What changes were proposed in this PR?
For partition table, local sort SI flow `current.segmentfile` is set in `CarbonSecondaryIndexRDD`
For the global sort, this value was not set. so, the main table query was resulting in 0 rows. Setting this value for global sort flow also.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4152/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2411/

File: hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java [hunk 1]
```diff
@@ -466,15 +466,15 @@ public IndexFilter getFilterPredicates(Configuration configuration) {
    * get data blocks of given segment
    */
   protected List<CarbonInputSplit> getDataBlocksOfSegment(JobContext job, CarbonTable carbonTable,
-      IndexFilter expression, List<Segment> segmentIds,
+      IndexFilter expression, List<Segment> validSegments,
```
Comment (kunal642) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java:469]: revert this change.. Not related to this PR

Comment (ajantha-bhat) [reply hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java:469]: Actually when the main table was giving 0 rows. I was debugging this flow. And this naming is wrong. segmentID should be string, here it is not the segmentID but the complete Segment object. so renamed. Still you want to revert it ?

Comment (kunal642) [reply hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java:469]: its fine.. ill merge

**Comment by kunal642** [comment]
LGTM
